src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -114,12 +120,16 @@ bool SoundRecorder::start(unsigned int sampleRate)
 ////////////////////////////////////////////////////////////
 void SoundRecorder::stop()
 {
-    // Stop the capturing thread
-    m_isCapturing = false;
-    m_thread.wait();
+    // Stop the recording if there is one
+    if (m_isCapturing)
+    {
+        // Stop the capturing thread
+        m_isCapturing = false;
+        m_thread.wait();
 
-    // Notify derived class
-    onStop();
+        // Notify derived class
+        onStop();
+    }
```

Comment:
The last two comments seem to be too much for me. It's clearly understandable from code what happens. ;) Also DRY, as the first comment already explains it all.